validations/src/validations.rs

```diff
@@ -1677,6 +1754,82 @@ pub fn validate_block_transactions(
         );
     }
 
+    // validate stake transactions in a block
+    let mut st_mt = ProgressiveMerkleTree::sha256();
+    let mut st_weight: u32 = 0;
+
+    // Check if the block contains more than one stake tx from the same operator
+    for i in 1..block.txns.stake_txns.len() {
+        let found = block.txns.stake_txns[i..].iter().find(|stake_tx| {
+            let stake_tx_aux = &block.txns.stake_txns[i - 1];
+
+            stake_tx.body.output.authorization.public_key
+                == stake_tx_aux.body.output.authorization.public_key
+        });
+
+        // TODO: refactor
+        if found.is_some() {
+            return Err(BlockError::RepeatedStakeOperator {
+                pkh: found
+                    .unwrap()
+                    .body
+                    .output
+                    .authorization
+                    .public_key
+                    .pkh()
+                    .hash(),
+            }
+            .into());
+        }
+    }
```

Comment:
I think this is iterating on the wrong dimension. I read this "for each stake transaction, fail if there is a later stake transaction that is conflicting with this". We could simply make an iterator on `block.txns.stake_txns`,  and use `.duplicates_by(|stake_txn| stake_txn.body.output.authorization.public_key).next()` from `itertools` to take an `Option<&StakeTransaction>`, or `.map(|stake_txn| stake_txn.body.output.authorization.public_key).duplicates().map(PublicKey::pkh).next()` if we only need `Option<&PublicKeyHash>.